docs/AUTOMATED_TESTS.md

```diff
@@ -1,12 +1,71 @@
 # Automated Tests
 
-DAPLink has an extensive automated test suite. It can be run on the current development environment or a release package. Test results are written to the current directory and printed to the console.
+DAPLink has an extensive automated test suite. It can be run on the current development environment (Windows only) or a release package (Windows, Mac or Linux). Test results are written to the current directory and printed to the console.
 
-Tests are launched by running ``python test/run_test.py`` in the Python virtual environment. The test logic discovers connected DAPLink boards and targets only those boards. Likewise, it detects what firmware you have built and targets only that firmware.
+Tests are launched by running ``python test/run_test.py`` in the Python virtual environment. The test logic discovers connected DAPLink boards and targets only those boards. Likewise, it detects what firmware you have built and targets only that firmware. Specify --help to view testing options (``run_test.py --help``).
 
 The tests rely on information in ``test/info.py``. If you are porting DAPLink to a new board, you will need to add to info.py. See [Adding a new board](PORT_BOARD.md) for details.
 
-### Two Areas of Testing
+## Preparing bootloader for Tests
+* Build the bootloader or interface project to be tested.
+* [Enable automation mode](ENABLE_AUTOMATION.md) on the board if it has not been enabled already.
+
+## Test on Windows
+### Prepare Host Machine
+Windows drive indexing needs to be turned off prior to running the tests.
+ 
+* Disable indexing on removable drives 
+ Run gpedit.msc (<Windows>+<R> to open run window) then go to:
+ Computer Configuration \ Administrative Templates \ Windows Components \ Search \ Do not allow locations on removable drives to be added to  libraries. Enable the setting there.
+ 
+### Running Tests
+ * Open CMD and cd to DAPLink directory.
+ * Activate virtual env, launch tests by running ``python test/run_test.py`
+   Specifying --firmwaredir is optional. For target images, either provide mbed.org authentication or use --targetdir to specify the location of pre-built target app binaries.
+ * Test results will be printed to console
+
+## Test on Linux
+###  Prepare Host Machine
+ Linux has a deamon called modemmanager that sends unsolicitated traffic to DAPLink. This will cause serial port tests to fail, typically at band rate initialization. The easiest way is to remove this deamon with ``sudo apt-get purge modemmanager``
+
+**Step 1.** Build Firmware Releases
+ The firmware images under test need to be built on Windows. Under tools directory, run ``build_release_uvision.bat``. Then copy the generated  uvision_release directory to Linux.
+ 
+**Step 2.** Install Tools
+ Install [Python 2, 2.7.9 or above](https://www.python.org/downloads/).
```

Comment:
Couldn't "Python 2, 2.7.9 or above" be replaced with "Python 2.7.9 or above"?

Reply:
Actually I think we do need to leave the 2 here, since Python 3 is not supported.